crates/nu-protocol/src/shell_error.rs

```diff
@@ -1056,6 +1056,13 @@ pub enum ShellError {
         #[label("Could not access '{column_name}' on this record")]
         span: Span,
     },
+
+    /// Operation interrupted by user
+    #[error("Operation interrupted by user")]
+    InterruptedByUser {
+        #[label("This operation was interrupted")]
+        span: Option<Span>,
+    },
```

Comment:
Mhh, my slash `ShellError` brain says could this use `IOInterrupted` so we don't need yet another variant. But I am surprised we don't have a `^C` specific error yet. (future goal would then be to move other stuff to use this variant if possible)

Reply:
Yeah, I think it would be good to finally add a `^C`-specific error. It's something I've wanted to use a few times now.

Reply:
(to clarify, I think this new `InterruptedByUser` variant is useful)

Are you OK landing this PR? Thank you for the detailed+helpful review!

Reply:
D'accord with the new variant. After addressing the time out "error" we can land